examples/stm32/nucleo-h563zi-make-baremetal-builtin/Makefile

```diff
@@ -2,18 +2,12 @@ CFLAGS  = -W -Wall -Wextra -Werror -Wundef -Wshadow -Wdouble-promotion
 CFLAGS += -Wformat-truncation -fno-common -Wconversion -Wno-sign-conversion
 CFLAGS += -g3 -Os -ffunction-sections -fdata-sections
 CFLAGS += -I. -Icmsis_core/CMSIS/Core/Include -Icmsis_h5/Include
-CFLAGS += -mcpu=cortex-m33 -mthumb -mfpu=fpv5-sp-d16 -mfloat-abi=hard
+CFLAGS += -mcpu=cortex-m33 -mthumb -mfpu=fpv5-sp-d16 -mfloat-abi=hard $(CFLAGS_EXTRA)
 LDFLAGS ?= -Tlink.ld -nostdlib -nostartfiles --specs nano.specs -lc -lgcc -Wl,--gc-sections -Wl,-Map=$@.map
 
-SOURCES = main.c syscalls.c sysinit.c
+SOURCES = main.c syscalls.c sysinit.c mongoose.c net.c packed_fs.c
 SOURCES += cmsis_h5/Source/Templates/gcc/startup_stm32h563xx.s # ST startup file. Compiler-dependent!
 
-# Mongoose-specific. See https://mongoose.ws/documentation/#build-options
-SOURCES += mongoose.c net.c packed_fs.c
-CFLAGS += -DMG_ENABLE_TCPIP=1 -DMG_ARCH=MG_ARCH_NEWLIB -DMG_ENABLE_CUSTOM_MILLIS=1
-CFLAGS += -DMG_ENABLE_CUSTOM_RANDOM=1 -DMG_ENABLE_PACKED_FS=1 
-CFLAGS += -DMG_ENABLE_DRIVER_STM32H=1 -DMG_OTA=MG_OTA_STM32H5 $(CFLAGS_EXTRA)
-
```

Comment:
I was thinking about this a lot.
What is the best place for keeping mongoose-related build options?

Is it Makefile or `mongoose_custom.h` ?
I'd say, it is `mongoose_custom.h`, because it is:
a. independent from the IDE. Make, Cube, whatever.
b. Thus can be easily copy-pasted from one project (e.g. baremetal) to another (e.g. keil)
c. it makes it very obvious what build options are. Unlike in makefile, where those are scattered around

Reply:
Wrt to point 'c': agreed, though we've been putting them all together.
Wrt points 'a' and 'b', unfortunately, that is not so.
Keil: The user sets the options using a configuration wizard. To make this work, special comments have to be added and comment-friendly options turned into usable Mongoose options later. Even though a plain mongoose_custom.h file coming from another project will probably work, the wizard will not. Plus, we always need to change MG_ARCH here, Keil uses ARM Compiler but it does not use newlib.
Cube: mongoose_custom is automatically generated from an Apache script. In a similar way to Keil, there are generic options that can be handled by the GUI that are later transformed to proper Mongoose macros. Even though a plain mongoose_custom.h file coming from another project will probably work, the Cube configurator will overwrite that file.

Anyway, I see your point, let's go that way. Just remember not to copy to Cube nor Keil... (and maybe not to MCUXpresso either...)